framework/src/loops/ComputeMortarFunctor.C

```diff
@@ -123,17 +130,38 @@ ComputeMortarFunctor::operator()()
 
     // Compute a JxW for the actual mortar segment element (not the lower dimensional element on
     // the secondary face!)
-    _subproblem.reinitMortarElem(msm_elem);
+    // fudge_factor is a convenient way of handling seconary elements with only partial overlapping
```

Comment:
```suggestion
    // fudge_factor is a convenient way of handling secondary elements with only partial overlap
```

Reply:
Thanks, I'm curious to know if you have any thoughts about this work around? 

I'm trying to think of some tests to make sure we still get the expected rate of convergence but its a bit tricky. For example, let's say we have 2 cubes next to each other with an equal value constraint on the 2 closest sides and the 2 cubes are slightly offset so that some of the elements on the secondary side aren't fully covered by elements on the primary side. If we use MMS, all the boundaries where the constraint isn't will essentially have fluxes fixed, but on those elements on the secondary side without  corresponding primary elements we'll be enforcing a 0 flux when they shouldn't be.

I can work on implementing this case more carefully (by actually forming the mortar segment mesh on these elements) but it'll complicate my implementation more so I'm hoping this is good enough.

Reply:
Just to close this discussion, the correct way to deal with mortar segments without a primary element across is to simply not assemble them. All has_primary logic has been removed and mortar segments without a primary element across are not included in the mortar segment mesh. This does create a conditioning issue when some elements only have very small contributions but will be addressed in the future with an update to dual shape functions calculations.